examples/next-forms/app/delete-form.tsx

```diff
@@ -0,0 +1,30 @@
+'use client'
+
+import { experimental_useFormState as useFormState } from 'react-dom'
+import { experimental_useFormStatus as useFormStatus } from 'react-dom'
+import { deleteTodo } from '@/app/actions'
+
+const initialState = {
+  message: null,
+}
+
+function DeleteButton({ message }: { message: string }) {
+  const { pending } = useFormStatus()
+
+  return (
+    <button type="submit" disabled={pending}>
+      {message ? message : 'Delete'}
+    </button>
+  )
+}
+
+export function DeleteForm({ id }: { id: number }) {
+  const [state, formAction] = useFormState(deleteTodo, initialState)
```

Comment:
Fun fact: You can also do `deleteTodo.bind(null, id)` either here or on the returned formAction. To pass the ID as an extra argument. It should preserve its ability to operate with progressive enhancement.

Gives you the same capabilities as closures but on the client. Using an actual closure on the client breaks progressive enhancement since now there's some client code running before.

Reply:
Which pattern would you prefer for us to recommend? I can update this if you feel `.bind()` is better.